File: test/lit/zig/struct_decl.zig
// RUN: %cot emit-cir %s | %FileCheck %s

// Feature #024: Struct declaration — Zig syntax

const Point = struct { x: i32, y: i32 };

const Color = struct { r: i8, g: i8, b: i8 };

pub fn takes_point(p: Point) i32 {
    return 0;
}

pub fn takes_color(c: Color) i32 {
    return 0;
}

// CHECK-LABEL: func.func @takes_point
// CHECK-SAME: !cir.struct<"Point", x: i32, y: i32>

// CHECK-LABEL: func.func @takes_color
// CHECK-SAME: !cir.struct<"Color", r: i8, g: i8, b: i8>

File: test/lit/ts/struct_decl.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// Feature #024: Struct declaration (interface) — TypeScript syntax
// Tests interface-as-type in function signatures (separate from struct_init)

interface Point {
    x: number;
    y: number;
}

interface Rect {
    left: number;
    top: number;
    width: number;
    height: number;
}

function takes_point(p: Point): number {
    return 0;
}

function takes_rect(r: Rect): number {
    return 0;
}

// CHECK-LABEL: func.func @takes_point
// CHECK-SAME: !cir.struct<"Point", x: i32, y: i32>

// CHECK-LABEL: func.func @takes_rect
// CHECK-SAME: !cir.struct<"Rect", left: i32, top: i32, width: i32, height: i32>
